src/client/opamConfigCommand.ml

```diff
@@ -638,10 +648,14 @@ let switch_allowed_fields, switch_allowed_sections =
           ("setenv", Modifiable (
               (fun nc c -> { c with env = nc.env @ c.env }),
               (fun nc c ->
+                 let open OpamTypes in
                  let env =
-                   List.filter (fun (vr,op,vl,_) ->
-                       None = OpamStd.List.find_opt (fun (vr',op',vl',_) ->
-                           vr = vr' && op = op' && vl = vl') nc.env) c.env
+                   List.filter (fun upd ->
+                       None = OpamStd.List.find_opt (fun upd' ->
+                           upd.envu_var = upd'.envu_var
+                           && upd.envu_op = upd'.envu_op
+                           && upd.envu_value = upd'.envu_value)
```

Comment:
```suggestion
                           String.equal upd.envu_var upd'.envu_var
                           && (upd.envu_op : OpamParserTypes.env_update_op) = upd'.envu_op
                           && String.equal upd.envu_value upd'.envu_value)
```

Reply:
I would also be more in favour of matching on the record to make sure that the compiler can give us a warning if a field is added to the record for example.

Something like:
```
let {envu_var; envu_op; envu_value;
     envu_comment = _; envu_separator = _; envu_format = _} = upd in
```